frontend/catalyst/utils/toml.py

```diff
@@ -264,15 +264,13 @@ def patch_schema1_collections(
 ):  # pylint: disable=too-many-arguments, too-many-branches
     """For old schema1 config files we deduce some information which was not explicitly encoded."""
 
-    # TODO: remove after PR #642 is merged in lightning
-    # NOTE: we mark GlobalPhase as controllables even if `quantum_control` flag is False. This
+    # We mark GlobalPhase as controllables even if `quantum_control` flag is False. This
     # is what actual device reports.
     if device_name == "lightning.kokkos":  # pragma: nocover
         native_gate_props["GlobalPhase"] = OperationProperties(
             invertible=False, controllable=True, differentiable=True
         )
 
-    # TODO: remove after PR #642 is merged in lightning
     if device_name == "lightning.kokkos":  # pragma: nocover
         observable_props["Projector"] = OperationProperties(
             invertible=False, controllable=False, differentiable=False
```

Comment:
Why is only the comment removed instead of the code block? Wasn't that the intention of the todo to remove the code block?

Reply:
If the toml is still incorrect for lightning.kokkos there should be another PR to lightning right?

Reply:
Both code fragments are now in the `patch_schema1_collections` function. It gets called for toml schema 1 only [here](https://github.com/PennyLaneAI/catalyst/pull/718/files/7800c69ebcebebf1b65d9fde2f1ba3c4851cf85a#diff-a87ceb7a61721c8cbd1c58d63193282cae87a0c3e8f8ea79c6ebbcde7a4770efR408). If I remember it correctly, both comments were created when this code resided in the common execution path.

Reply:
But the fact that we patch this gate into the toml file doesn't that mean that the upstream toml file is out of date?

Reply:
Wait these patches are for schema 1 only? Kokkos is already on schema 2, so why not delete these patches?

Reply:
Yes they are. Schema 2 is in PLL master, but `pip install` installs Pennylane-lightning with old schema 1.

Reply:
Hmm. Actually, this particular patch (adding Projector gate) is really outdated so I am removing it. Schema 1 already contains this gate.

But the second patch (discussed above) fixes flags in a way that could not be represented in schema1. So it is actually required.

Reply:
David, here is the related [conversations](https://github.com/PennyLaneAI/pennylane-lightning/pull/642#discussion_r1535478642) in PL 642.  Both `C(GlobalPhase)` and `Projector` are not supported by Kokkos C++, but they are supported by its Python API.
* Projector patch is outdated because we relaxed the validation for observables: we allow devices to have Python-only observables.
* For C(GlobalPhase) we do not have a way to represent it even in schema 2. So the patch is needed and more, it has to be moved to the common execution pipeline.

Reply:
I don't think the validation is good. It is a crutch for us as developers not to forget to update the lightning toml files when their device capabilities change, but has become obsolete with the new device api (no more `.operations` class property) and causes too much confusion and issues when the Python device and C++ device diverge.

Reply:
Well, the original purpose of validation was to solve the weird situation where one team develops APIs and another team (we) is responsible for the correct API description. If the validation is no longer possible (no .operators) - then indeed there is no use for it, but this means that we must resolve the situation in some other way.

Reply:
David, thank you for checking this story.